### ISSUE TITLE
Can't import systemjs from npm with Meteor 1.3-modules-beta.0

### ISSUE THREAD

**Issue by stphdenis** [meteor/meteor#5828]
After the creation of new fresh app followed by the install of systemjs via npm :
```
meteor --release 1.3-modules-beta.0 create importSystemjs
cd importSystemjs
npm install systemjs
```
and adding a `require('systemjs');`, i get the following error: 
```
ReferenceError: dirPath is not defined
    at ImportScanner._tryToResolveImportedPath (C:\tools\isobuild\import-scanner.js:273:27)
    at C:\tools\isobuild\import-scanner.js:96:36
```

**Comment by bySabi** [comment]
I got the same error: https://forums.meteor.com/t/meteor-1-3-early-beta-now-available/14723/87

**Comment by stphdenis** [comment]
This error is fixed by this patch :
https://github.com/meteor/meteor/commit/fb73d9e3704556018310b7c873b4e7a95fac807c

**Comment by stphdenis** [comment]
After applying https://github.com/meteor/meteor/commit/fb73d9e3704556018310b7c873b4e7a95fac807c, an other error come's up :
```
C:\Users\SDenis\meteor\importSystemjs\.meteor\local\build\programs\server\app\app.js:134
   "_args": [
          ^
 SyntaxError: Unexpected token :
     at C:\Users\...\meteor\importSystemjs\.meteor\local\build\programs\server\boot.js:241:30
```
The file `app/node_modules/systemjs/package.json` is included in `app.js` as if it was a javascript one.

**Comment by rozzzly** [comment]
Check out universe-modules, more specifically universe-modules-entrypoint. Its the same babel compiler that meteor uses, but with more babel transforms enabled. And it uses system js. I've been using it and am pretty satisfied. Doesn't have effortless npm install yet (just has a npm.json you can list npm pkgs in). All of that gets pushed through systemjs as well.

I mention this because meteor modules and system js have a lot of overlap in functionality. I havent tried the new meteor modules pkg yet, waiting til 1.3, but will deff have to run some pref tests. I'm not sure that I will make the switch because I like being able to customize babel/the loader/all of systemjs's 'map', custom loaders, etc stuff as well.

Plus its using the es6 spec. @benjamn greatwork, you've got some pretty solid code here. But I got to say, I think that not strictly adhering to the es6 system spec was a bad decision. Eventually that's gonna be native, and meteor will have yet another outdated boondoggle. It wouldn't be hard to turn this into a polyfill, but systemjs supports more module types, is much more mature, etc. I remember seeing a post referencing Systemjs's System.register not being so great, but beyond that I don't see why this was custom made. Just because less overhead/filesize? Does compatibility with meteors setup require a loader which cant be extended from systemjs? Or is there something else that im missing.

**Comment by stubailo** [comment]
> I think that not strictly adhering to the es6 system spec was a bad decision.

Curious, where are we not adhering to specs?

**Comment by benjamn** [comment]
@stphdenis both of these bugs will be fixed in the next beta release. There's going to be a more formal announcement about the new release soon, but you should be able to do `meteor --release 1.3-modules-beta.1 create importSystemJs` as of now.

**Comment by benjamn** [comment]
For what it's worth, the `Reflect.Loader` and `System` APIs were removed from the ECMAScript specification in October 2014 ([Draft Rev 28](http://wiki.ecmascript.org/doku.php?id=harmony:specification_drafts)), and now live on in a separate spec: https://whatwg.github.io/loader/

When that spec has any native implementations (preferably more than one!), we'll consider switching our bundle format to use `System.register`. Until then, `meteorInstall` is serving our purposes well, not least because it supports CommonJS `require` and `exports` syntax, whereas `System.register` is designed to work only with `import` and `export` statements. For better or worse, easy CommonJS support is critical for interoperating with today's npm ecosystem.

**Comment by bySabi** [comment]
`1.3-beta1` solved 'my' problem when I try to add `npm` local package.
Thanks!!!

**Comment by benjamn** [comment]
Fixed in 1.3b1: https://github.com/meteor/meteor/issues/5788#issuecomment-165935104

**Comment by rozzzly** [comment]
@benjamn 

> ... the Reflect.Loader and System APIs were removed from the ECMAScript specification in October 2014 (Draft Rev 28) ...

Touché. I was unaware of this. And fair enough. Like I said this is good work, im just quibbling over internals. 

----

@stubailo 

> Curious, where are we not adhering to specs?

Let me rephrase, _the System loader spec as seen [here](https://whatwg.github.io/loader/), not es6 modules. This **does** handle `import`/`export`, but  not the specific internal semantics for resolution, fetching, etc._

**Comment by sammkj** [comment]
I'm currently facing a problem importing SystemJS with beta.2. It gives the following error:
```
index.js:8  Uncaught TypeError: Cannot read property 'match' of undefined
```
which points to 
```
var isWindows = process.platform.match(/^win/);
```
